src/index.ts

```diff
@@ -621,37 +647,8 @@ export class TOTPStrategy<User> extends Strategy<User, TOTPVerifyParams> {
     // Verify TOTP (@epic-web/totp).
     const isValid = verifyTOTP({ ...totp, otp })
     if (!isValid) {
-      await this.handleTOTP(sessionTotp, { attempts: dbTOTP.attempts + 1 })
+      await this.updateTOTP(sessionTotp, { attempts: dbTOTP.attempts + 1 }, expiresAt)
       throw new Error(this.customErrors.invalidTotp)
     }
   }
-
-  private async _handleExpiresAt(
```

Comment:
Happy to have this out of the code; it was somewhat of a headache to come up with a solution that could be beneficial in most cases, based on the provided user database.

I'm not sure if it was the best implementation, but it did its job in the meantime.